src/mono/wasi/runtime/main.c

```diff
@@ -21,8 +21,38 @@ int main(int argc, char * argv[]) {
 #endif
 	mono_wasm_load_runtime("", 0);
 
+	int arg_ofs = 0;
+#ifdef WASM_SINGLE_FILE
+	/*
+	 * For single-file bundle, running with wasmtime:
+	 *
+	 *  $ wasmtime run --dir . MainAssembly.wasm [args]
+	 *
+	 * arg0: MainAssembly
+	 * arg1-..: args
+	 */
 	const char* assembly_name = dotnet_wasi_getentrypointassemblyname();
 	MonoAssembly* assembly = mono_assembly_open(assembly_name, NULL);
+#else
+	/*
+	 * For default case which uses dotnet.wasm, running with wasmtime:
+	 *
+	 *  $ wasmtime run --dir . dotnet.wasm MainAssembly [args]
```

Comment:
Shouldn't we have a unified way to run the application? For example, could we rename the `dotnet.wasm` to `MainAssembly.wasm` even for non-single file bundle?

Reply:
I think that for non-single application, the interesting feature is that dotnet.wasm is always the same content. Full set of features, with which you could run any managed assembly.